overlord/snapstate/snapstate.go

```diff
@@ -511,20 +511,29 @@ func defaultContentPlugProviders(st *state.State, info *snap.Info) []string {
 // validateFeatureFlags validates the given snap only uses experimental
 // features that are enabled by the user.
 func validateFeatureFlags(st *state.State, info *snap.Info) error {
-	if len(info.Layout) == 0 {
-		return nil
-	}
-
 	tr := config.NewTransaction(st)
-	var featureFlagLayouts bool
-	if err := tr.GetMaybe("core", "experimental.layouts", &featureFlagLayouts); err != nil {
-		return err
+
+	if len(info.Layout) > 0 {
+		var flag bool
+		if err := tr.GetMaybe("core", "experimental.layouts", &flag); err != nil {
+			return err
+		}
+		if !flag {
+			return fmt.Errorf("cannot use experimental 'layouts' feature, set option 'experimental.layouts' to true and try again")
```

Comment:
I realize this is just copying over from the prior error message, but can we please use something like this:

`experimental feature disabled - test it setting "experimental.layouts" to true`

That's shorter, points out that this is about _testing_ the feature, and avoids the duplicated nature of the original messages.